dbms/src/Columns/ColumnVector.cpp

```diff
@@ -73,25 +74,47 @@ void ColumnVector<T>::getPermutation(bool reverse, UInt64 limit, int nan_directi
 {
     size_t s = data.size();
     res.resize(s);
-    for (size_t i = 0; i < s; ++i)
-        res[i] = i;
 
     if (limit >= s)
         limit = 0;
 
     if (limit)
     {
+        for (size_t i = 0; i < s; ++i)
+            res[i] = i;
+
         if (reverse)
             std::partial_sort(res.begin(), res.begin() + limit, res.end(), greater(*this, nan_direction_hint));
         else
             std::partial_sort(res.begin(), res.begin() + limit, res.end(), less(*this, nan_direction_hint));
     }
     else
     {
-        if (reverse)
-            pdqsort(res.begin(), res.end(), greater(*this, nan_direction_hint));
+        if constexpr ((std::is_signed_v<T> || std::is_unsigned_v<T>) && !std::is_same_v<T, UInt128>)
```

Comment:
It's better to check this in the beginning of the function and return to save file history:
if (!limit)
  if constexpr ((std::is_signed_v<T> || std::is_unsigned_v<T>) && !std::is_same_v<T, UInt128>)
  {
    sortWithRadix(...);
    return;
  }

Reply:
It's fine.